### DISCUSSION TITLE
About AuthPage and AuthGuard

### DISCUSSION THREAD

**Discussion by jitendravn** [nylo-core/nylo#130] [Ideas] [answered]
Hello Sir,
In Nylo we have authpage and routeguard
we can add in our router file is ok.
what when use don't use routeTo() for navigating

I have the bottom navigation bar
and some menus are auth-restricted
but I'm not able to use bottom navigation pages as authPage: true

assume  I have this route
```Dart
router.route(
        ChatPage.path,
        (context) => ChatPage(),
        transition: PageTransitionType.fade,
        authPage: false,
        routeGuards: [
          AuthRouteGuard(),
        ],
);
```

but the route I'm not using with routeTo(ChatPage.path)

I'm Directly passing this in the bottom navigation bar 

So If possible manage Auth in NyState/statelesswidget page
like 
bool get isAuthPage =>true/false

**Comment by agordn52** [comment, marked as answer]
Hi @jitendravn,

`RouteGuard`'s are used to protect a page on navigation E.g. using the `routeTo` helper.

`authPage` is used to override the initial page to be shown if the user is authenticated in your project.

If you need to handle user permissions for authenticated users, check out this package [permission_policy](https://pub.dev/packages/permission_policy).
